### ISSUE TITLE
Too many requests

### ISSUE THREAD

**Issue by larrelandin** [TekniskSupport/home-assistant-resrobot#30]
Hey, I have had issues for quite a while now. The number of allowed requests to Trafiklab is going through the roof every single month. And after a while I get no updates.
![image](https://github.com/user-attachments/assets/a0b283c2-7f04-4029-ba20-f40641a38d2e)

Any ideas why this is happening?

Here is my config:
```
sensor:
  - platform: resrobot
    fetch_interval: 10
    key: !secret trafiklab
    departures:
      - stop_id: 740072471
        name: "Buss till Botulfsplatsen"
        max_journeys: 20
        sensors: 3
        unit: "🕑"
        time_format: "%H:%M"
        filter:
          - line: 5
            type: "contains"
            direction: "annehem"
      - stop_id: 740016732
        name: "Bussar från Gastelyckan Ö"
        max_journeys: 20
        sensors: 6
        unit: "🕑"
        time_format: "%H:%M"
        filter:
          - direction: "Lund C"
      - stop_id: 740016981
        name: "Bussar från Botulfsplatsen"
        max_journeys: 20
        sensors: 3
        unit: "🕑"
        time_format: "%H:%M"
        filter:
          - line: 5
            direction: "Lund Råbylund"
      - stop_id: 740016757
        name: "Bussar från Bankgatan"
        max_journeys: 20
        sensors: 2
        unit: "🕑"
        time_format: "%H:%M"
        filter:
          - line: 161
          - line: 162

```

**Comment by TekniskSupport** [comment]
Hi,

I would suggest changing the fetch interval to a higher value, or try removing a departure.

With 1 departure and no reboots it would do about 4320 requests a month (30 days * 24 hours * 60 minutes)/10 minutes;

I have been thinking about the limitations when i built the integration, that's the reason i use helper entities to store the information, instead of each sensor doing it's own request.

But let's do some calculations with this formula
 (minutes a month / fetch interval) * number of helpers (departures)
 
This alone should not reach the limit of 25k, but if you have more sensors not shown in the issue, or another instance of home assistant with the same api key

Maybe you still have devices that you used when setting it up, somewhere not being used in the UI?
 
It performs a request to the API for each departure after every boot, and after that every 10 minutes with the current configuration. It should add up to about 17280 with your 4 departures, reboots excluded.

Another solution could involve using another api key. That could be against the trafficlab end user license agreement. so maybe not a recommended solution..

```yaml
sensor:
  - platform: resrobot
    fetch_interval: 10
    key: !secret trafiklab
    departures:
      - stop_id: 740072471
        name: "Buss till Botulfsplatsen"
        max_journeys: 20
        sensors: 3
        unit: "🕑"
        time_format: "%H:%M"
        filter:
          - line: 5
            type: "contains"
            direction: "annehem"

sensor:
  - platform: resrobot
    fetch_interval: 10
    key: !secret trafiklab2
    departures:
      - stop_id: 740016757
        name: "Bussar från Bankgatan"
        max_journeys: 20
        sensors: 2
        unit: "🕑"
        time_format: "%H:%M"
        filter:
          - line: 161
          - line: 162
 ```

**Comment by larrelandin** [comment]
AFAIK there is no other configuration and no hidden sensors, there is no other instances using this same API. I'll try to increase the fetch_interval to 30 but I am pretty sure that is not the case.
Anyway to add some kind of debug flag for the integration to get extended log files?

**Comment by TekniskSupport** [comment]
Not at the time, but i can add one when i have a minute left over

**Comment by TekniskSupport** [comment]
I have added a new optional configuration option called debug_mode
which is a True/False value and logs each time the integration updates information from the API

**Comment by larrelandin** [comment]
Thanks for this, I have tested debug_mode and even though I have _fetch_interval: 300_ I still get a log every minute like this:
_2024-12-11 15:26:09.319 WARNING (MainThread) [custom_components.resrobot.sensor] ResRobot Update_

**Comment by TekniskSupport** [comment]
That's interesting, mine seems to be working pretty much as intended

![image](https://github.com/user-attachments/assets/8e10cc80-b566-4d6f-9362-4944a952ee7e)


What is the output from:
`find . -type f 2>/dev/null| grep -E '\.yml|\.yaml' |xargs cat|grep resrobot -A10`

(or if you have yq installed:)
`find . -iname '*.yaml' 2>/dev/null -exec yq  '(.[] | select(.platform == "resrobot"))' {} \; `

**Comment by larrelandin** [comment]
Ok, I have managed to get a bit closer to the issue here, the integration calls an update each minute no matter what I set the fetch_interval to. However, if I disable the third stop as indicated below, it seem to work!?

```
sensor:
  - platform: resrobot
    debug_mode: true
    fetch_interval: 30
    key: !secret trafiklab
    departures:
      - stop_id: 740072471
        name: "Buss till Botulfsplatsen"
        max_journeys: 20
        sensors: 3
        unit: "🕑"
        time_format: "%H:%M"
        filter:
          - line: 5
            type: "contains"
            direction: "annehem"
      - stop_id: 740016732
        name: "Bussar från Gastelyckan Ö"
        max_journeys: 20
        sensors: 6
        unit: "🕑"
        time_format: "%H:%M"
        filter:
          - direction: "Lund C"
         *** Disabled ***
      - stop_id: 740016981
        name: "Bussar från Botulfsplatsen"
        max_journeys: 20
        sensors: 3
        unit: "🕑"
        time_format: "%H:%M"
        filter:
          - line: 5
            direction: "Lund Råbylund"
        *** END ***
      - stop_id: 740016757
        name: "Bussar från Bankgatan"
        max_journeys: 20
        sensors: 2
        unit: "🕑"
        time_format: "%H:%M"
        filter:
          - line: 161
          - line: 162
```

**Comment by TekniskSupport** [comment]
I tested replicating your config, and found that 

2024-12-12 18:08:55.536 WARNING (MainThread) [custom_components.resrobot.sensor] ResRobot Update
2024-12-12 18:08:55.938 ERROR (MainThread) [custom_components.resrobot.sensor] ResRobot found no trips
2024-12-12 18:09:55.512 WARNING (MainThread) [custom_components.resrobot.sensor] ResRobot Update
2024-12-12 18:09:56.109 ERROR (MainThread) [custom_components.resrobot.sensor] ResRobot found no trips
2024-12-12 18:10:55.538 WARNING (MainThread) [custom_components.resrobot.sensor] ResRobot Update
2024-12-12 18:10:55.890 ERROR (MainThread) [custom_components.resrobot.sensor] ResRobot found no trips

The sensor you have commented out matches no trips, and when that happens it seems like it is retrying again and again

Thank you for your help!
I will make adjustments to keep this from happening

**Comment by TekniskSupport** [comment]
@larrelandin now fixed in version v0.0.20!
